plugin-server/src/worker/ingestion/event-pipeline/pluginsProcessEventStep.ts

```diff
@@ -5,22 +5,48 @@ import { runProcessEvent } from '../../plugins/run'
 import { droppedEventCounter } from './metrics'
 import { EventPipelineRunner } from './runner'
 
+// Some properties are internal, which plugins shouldn't be able to change
+// We don't want these to be passed to the processEvent plugins
+const hiddenProperties = [
+    '$heatmap_data', // Is processed and removed at a later pipeline step
+    '$elements', // This is deprecated
+    '$groups', // To avoid dependencies if we want to change how we use this property
+    '$active_feature_flags', // To avoid dependencies if we want to change how we use this property
```

Comment:
I take it that you just know these aren't used in any active plugins?

Reply:
Better safe than sorry (good call out), so double checking:
Code:
- `$heatmap_data` - is new so good
- `$elements` - is mentioned in code by 6 plugins 4 destinations [metabase](http://metabase-prod-us/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJzZWxlY3QgKiBmcm9tIHBvc3Rob2dfcGx1Z2luIHdoZXJlIGlkIGluIChcbnNlbGVjdCBwbHVnaW5faWQgZnJvbSBwb3N0aG9nX3BsdWdpbnNvdXJjZWZpbGUgd2hlcmUgc291cmNlIExJS0UgJyUkZWxlbWVudHMlJ1xuKSIsInRlbXBsYXRlLXRhZ3MiOnt9fSwiZGF0YWJhc2UiOjM0fSwiZGlzcGxheSI6InRhYmxlIiwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnt9fQ==), which get no changes from this, property_flatener that is already handled by linked PR and netdata, which I thought was done, but I'm wrong https://github.com/PostHog/posthog/issues/22020 so I'll remove elements from here
- $groups - only property flattener processEvent (denylisting there explicitly) 
- $active_feature_flags - only property flattener processEvent (denylisting there explicitly) 

Secondly folks could be using this within their configs:
- pluginConfigs via equivalent of `select * from posthog_pluginconfig where config::text like '%$groups%'`: there's a single user in EU setting `$active_feature_flags` to null via propertyFilter, they aren't exporting it though, so only within posthog, asked about this in https://posthog.slack.com/archives/C034XD440RK/p1714584381832039

Thirdly folks could be using it within plugin config attachments, don't know how to check these effectively, so I'm going to just go with `$heatmap_data` only for now to be safe.